### ISSUE TITLE
Error in 1deg_core_nyf, get_field_dim_lens: Unsupported dimension name

### ISSUE THREAD

**Issue by aidanheerdegen** [OceansAus/access-om2#33]
Abhishek is trying to run the access-om2 1 deg CORE NYF experiment and is getting an inscrutable netCDF error (I think)

```
get_field_dim_lens: Unsupported dimension name
```

The access-om2.out file is at this point when it happens:

 ```
MOCN: _get_localcomm_ OK! il_commlocal=            3
 Reading setup_nml
 Reading grid_nml
 Reading tracer_nml
 Reading thermo_nml
 Reading dynamics_nml
 Reading shortwave_nml
 Reading ponds_nml
 Reading forcing_nml
 Diagnostic output will be in file 
 ice_diag.d                                                                     
  
 Reading zbgc_nml
```

I don't think the traceback is useful, as it is just getting a kill signal when the error is tripped, but here is the first one in any case:
```forrtl: error (78): process killed (SIGTERM)
Image              PC                Routine            Line        Source             
fms_ACCESS-OM_030  0000000001A89F11  Unknown               Unknown  Unknown
fms_ACCESS-OM_030  0000000001A8804B  Unknown               Unknown  Unknown
fms_ACCESS-OM_030  0000000001A36074  Unknown               Unknown  Unknown
fms_ACCESS-OM_030  0000000001A35E86  Unknown               Unknown  Unknown
fms_ACCESS-OM_030  00000000019B7139  Unknown               Unknown  Unknown
fms_ACCESS-OM_030  00000000019C157C  Unknown               Unknown  Unknown
libpthread-2.12.s  00002B9FBF0C37E0  Unknown               Unknown  Unknown
libmlx4-rdmav2.so  00002B9FCCC06673  Unknown               Unknown  Unknown
mca_btl_openib.so  00002B9FCA6D16A2  Unknown               Unknown  Unknown
mca_btl_openib.so  00002B9FCA6DA3F5  Unknown               Unknown  Unknown
mca_btl_openib.so  00002B9FCA6DA814  Unknown               Unknown  Unknown
mca_btl_openib.so  00002B9FCA6DA8AC  Unknown               Unknown  Unknown
libopen-pal.so.13  00002B9FC308A66C  opal_progress         Unknown  Unknown
libmpi.so.12.0.2   00002B9FBEB50DFC  Unknown               Unknown  Unknown
libmpi.so.12.0.2   00002B9FBEB5132B  ompi_request_defa     Unknown  Unknown
mca_coll_tuned.so  00002B9FCB9AE106  Unknown               Unknown  Unknown
mca_coll_tuned.so  00002B9FCB9AE737  ompi_coll_tuned_b     Unknown  Unknown
mca_coll_tuned.so  00002B9FCB9A1888  ompi_coll_tuned_b     Unknown  Unknown
libmpi.so.12.0.2   00002B9FBEB6B1A8  MPI_Barrier           Unknown  Unknown
libmpi_mpifh.so.1  00002B9FBE8BA2C9  Unknown               Unknown  Unknown
fms_ACCESS-OM_030  0000000001795CFB  mpp_mod_mp_mpp_in         155  mpp_util_mpi.inc
fms_ACCESS-OM_030  0000000001457669  fms_mod_mp_fms_in         336  fms.F90
fms_ACCESS-OM_030  000000000040FFAA  MAIN__                    222  ocean_solo.F90
fms_ACCESS-OM_030  000000000040D91E  Unknown               Unknown  Unknown
libc-2.12.so       00002B9FBF2EFD1D  __libc_start_main     Unknown  Unknown
fms_ACCESS-OM_030  000000000040D829  Unknown               Unknown  Unknown
```

**Comment by marshallward** [comment]
This says it's failing on MPI_init(), maybe a trace from a different rank
would be more useful here?

**Comment by aidanheerdegen** [comment]
That was the first one, they are all pretty similar. The only one that is different is this:
```forrtl: error (78): process killed (SIGTERM)
Image              PC                Routine            Line        Source             
fms_ACCESS-OM_030  0000000001A89F11  Unknown               Unknown  Unknown
fms_ACCESS-OM_030  0000000001A8804B  Unknown               Unknown  Unknown
fms_ACCESS-OM_030  0000000001A36074  Unknown               Unknown  Unknown
fms_ACCESS-OM_030  0000000001A35E86  Unknown               Unknown  Unknown
fms_ACCESS-OM_030  00000000019B7139  Unknown               Unknown  Unknown
fms_ACCESS-OM_030  00000000019C157C  Unknown               Unknown  Unknown
libpthread-2.12.s  00002B3C917E67E0  Unknown               Unknown  Unknown
fms_ACCESS-OM_030  00000000019B4A71  Unknown               Unknown  Unknown
fms_ACCESS-OM_030  0000000001A20F8F  Unknown               Unknown  Unknown
fms_ACCESS-OM_030  0000000001A1FDCB  Unknown               Unknown  Unknown
fms_ACCESS-OM_030  0000000001794A84  mpp_mod_mp_mpp_in        1322  mpp_util.inc
fms_ACCESS-OM_030  0000000001457669  fms_mod_mp_fms_in         336  fms.F90
fms_ACCESS-OM_030  000000000040FFAA  MAIN__                    222  ocean_solo.F90
```

**Comment by marshallward** [comment]
Can you count the number of available traces vs number of ranks?  Maybe the failed netcdf call exited gracefully.  (I can also have a look if you're busy)

**Comment by aidanheerdegen** [comment]
There are 304 cpus in the job and 

```
grep forrtl access-om2.err | wc -l                                                                                                      
264
```

**Comment by marshallward** [comment]
If it's reproducible, you can try adding `-tag-output` to mpiexec to see which rank is producing the netCDF errors (though I suspect it's rank 0).

I might be chasing this the wrong way though, probably easier to backtrack through FMS.

**Comment by aidanheerdegen** [comment]
I find it passing strange that google comes up with a big fat zero when searching for `get_field_dim_lens`

**Comment by marshallward** [comment]
I find it doubly weird that neither netcdf nor HDF5 appears to have and `get_field_dim_*` symbols in their tables.  And no reference in any of my MOM executables either.  WTF is this error??

**Comment by nicjhan** [comment]
I'm guessing that this is something to do with the runoff. I'll take a look at it now too. 

I think in general we have a big problem with error messages. There are so many different places that these can come from. I've added an issue here: https://github.com/OceansAus/access-om2/issues/36

**Comment by nicjhan** [comment]
You can blame me for this terrible error message. git blame is showing that I either wrote or modified some code in the cpl_netcdf_setup model in MATM.

**Comment by nicjhan** [comment]
This experiment won't run until I've fixed: https://github.com/OceansAus/access-om2/issues/37

**Comment by aidanheerdegen** [comment]
Thanks Nic. Odd that google couldn't find that string.

**Comment by nicjhan** [comment]
Fixed by #37
